courses/fundamentals_of_ada/adv_270_subprogram_contracts.rst

```diff
@@ -466,64 +449,55 @@ Example for Attribute 'Old
 
       procedure Shift_And_Advance (Index : in out Integer) with Post =>
          -- call At_Index before call
-         At_Index (Index)'Old
+         Global (Index)'Old
             -- look at Index position in Global before call
             = Global'Old (Index'Old)
          and
          -- call At_Index after call with original Index
-         At_Index (Index'Old)
+         Global (Index'Old)
             -- look at Index position in Global after call
             = Global (Index);
 
--------------------------------------
-What Happens When 'Old Is Evaluated
--------------------------------------
+------------------------------------------------
+Error on conditional Evaluation of :ada:`'Old`
+------------------------------------------------
 
-* Copy made on entrance for use by postconditions
-* "Safety" checks in postcondition weren't applied to the entrance copy evaluation
-* Incorrect
+* This code is **incorrect**
 
-      .. code:: Ada
+.. code:: Ada
 
-          procedure Clear_Character (In_String : in out String;
-                                     Look_For  : in     Character;
-                                     Found_At  :    out Integer)
-             with Post => Found_At in In_String'Range and
-                          In_String (Found_At'Old) = Look_For;
+  procedure Clear_Character (In_String : in out String;
+                             At_Position : Positive)
+     with Post => (if Found_At in In_String'Range
+                   then In_String (Found_At)'Old = ' ');
 
-   - On entry, `Found_At` is not valid, so :ada:`In_String(Found_At'Old)` will likely raise an exception
+* Copies :ada:`In_String (Found_At)` on entry
 
-* Solution (required)
+   - Will raise an exception on entry if :ada:`Found_At not in In_String'Range`
```

Comment:
Missing tick after "Found_At"

Reply:
Sorry I don't understand what you mean
`Found_At not in InString'Range` is a proper Ada membership test